<HTML>
<HEAD>
  <META NAME="GENERATOR" CONTENT="Adobe PageMill 2.0 Mac">
  <TITLE>Evolution of the RecA protein (by Jonathan A. Eisen)</TITLE>
</HEAD>
<BODY>

<H2 ALIGN=CENTER>Evolution of RecA</H2>

<H3>Some interesting features of the evolution of RecA proteins:</H3>

<UL>
  <LI><I>Arabidopsis thaliana</I> encodes a nuclear gene for a RecA protein
  that functions in the chloroplast and is evolutionarily related to cyanobacterial
  RecA genes, suggesting that it was transfered to the nucleus of plants
  sometime after the endosymbiosis with the chloroplast evolved.<BR>
  
  <LI>Many Mycobacterial species have inteins (protein introns which are
  removed post-translationally) in their RecA proteins (see <BR>
  <A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11034333&dopt=Abstract">Saveset
  al 2000 </A>for review).<BR>
  
  <LI><I>Myxococcus xanthus</I> encodes two RecAs genes (see <A HREF="http://www.ncbi.nlm.nih.gov/htbin-post/Entrez/query?db=m&form=6&uid=95332233&dopt=r">Norioka
  et al.1995</A>)<BR>
  
  <LI><I>Lactococcus lactis</I> encodes an usual RecA-like protein on the
  phage resistance plasmid pNP40. This gene is next to a UmuC-like gene.
  The functions of these proteins are not well characterized.<BR>
  
  <LI><I>Spiroplasma citri </I>has a <I>recA</I>-like gene but it only has
  ~390 bp similar to other <I>recAs</I>. <I>S. melliferum</I> encodes a complete
  <I>recA</I> gene but it contains a stop codon early in the gene. Both species
  are highly UV sensitive.<BR>
  
  <LI>The pea-aphid endosymbiont <I>Buchnera aphidicola</I> does not encode
  a recA homolog in its complete genome (<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=10993077&dopt=Abstract">Shigenobu
  et al, 2000)</A>
</UL>

<H3>Use of RecA as a phylogenetic marker:</H3>

<UL>
  <LI><A HREF="http://www.ncbi.nlm.nih.gov/htbin-post/Entrez/query?uid=94141940&form=6&db=m&Dopt=r">Lloyd
  and Sharp</A> (1993) first addressed the possibility of using phylogenetic
  analysis of <I>recA</I> genes as a tool to study bacterial evolutionary
  relationships.<BR>
  
  <LI>I followed up their analysis using the sequences of 40 additional <I>recA</I>
  genes that have been entered into the databases through 1995. (<A HREF=
  "http://www.ncbi.nlm.nih.gov/htbin-post/Entrez/query?uid=96139045&form=6&db=m&Dopt=b">Eisen,
  1995</A>). In this study I showed that RecA based phylogenies are very
  similar to rRNA based phylogenies from the same species and that RecA trees
  have similar powers of resolution. For more detail about this study see
  <A HREF="RecA.Eisen1995.html" TARGET="content">Notes on Eisen 1995<BR>
  </A>
  <LI>Tanja Gruber and I used RecA comparisons to examine the phylogeny of
  species in the <I>Chlorobium</I> and <I>Chloroflexus</I> genera (<A HREF=
  "http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=9595663&dopt=Abstract">Gruber
  et al 1998) </A>For more detail about this see <A HREF="RecA.Gruber1998.html">Notes
  on Gruber et al 1998</A>.<BR>
  
  <LI>Advantages of using RecA as a phylogenetic marker for bacteria
  <UL>
    <LI>Relatively easy to clone for a protein encoding gene
    <LI>Conserved function among bacteria
    <LI>Sequence conservation varies across molecule
    <LI>Alignments are essentially unambiguous
    <LI>Sequence information can be helpful in creating<I> recA-</I> mutants
    <LI>3D crystal structure of <I>E. coli</I> RecA available<BR>
    
  </UL>
  <LI>Disadvantages of using RecA as a phylogenetic marker for bacteria
  <UL>
    <LI>Multiple divergent genes in at least one species (<I>M. xanthus</I>)
    <LI>Number of sequences still 100 fold lower than for SS-rRNA
    <LI>Similarity to Archaeal and Eukaryotic proteins somewhat low<BR>
    
  </UL>
  <LI>Other studies using RecA sequences for phylogenetic or evolutionary
  studies of bacteria
  <UL>
    <LI><A HREF="http://www.ncbi.nlm.nih.gov/htbin-post/Entrez/query?uid=9311137&form=6&db=m&Dopt=b">Kullen,
    et al.</A> Phylogeneetic analysis of bifidobacteria using recA
    <LI><A HREF="http://www.ncbi.nlm.nih.gov/htbin-post/Entrez/query?uid=7649456&form=6&db=m&Dopt=b">Nowak,
    et al. 1995</A>, <A HREF="http://www.ncbi.nlm.nih.gov/htbin-post/Entrez/query?uid=9025077&form=6&db=m&Dopt=b">Nowak
    , et al. 1996</A>. Genotyping of acinetobacters using recA PCR-RFLPs.
    <LI>Van Waasbergen et al. 1996. Abstracts of ASM 96: 404. Phylogeny of
    deep subsurface <I>Arthrobacter</I>.
    <LI><A HREF="http://www.ncbi.nlm.nih.gov/htbin-post/Entrez/query?uid=8995060&form=6&db=m&Dopt=b">Feil,
    et al.</A> recA sequences to study recombination and genetic exchange in
    <I>Neisseria</I>.
    <LI><A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=10698993&dopt=Abstract">Vandamme
    et al 2000 </A>used recA sequences to study the population structure of
    <I>Burkholderia stabilis</I>
    <LI><A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=10608776&dopt=Abstract">Kuipers<I>
    </I>et al 2000 </A>used analysis of RecA diversity within <I>H. pylori
    </I>to study population structure
    <LI><A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=10368955&dopt=Abstract">Smith
    et al 2000 </A>used analysis of RecA to study population structure of <I>Neisseria</I>
    <LI><A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=10024551&dopt=Abstract">Byun
    et al 2000</A> used analysis of RecA to study evolution of pathogenic <I>Vibrios</I>
    <LI><A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=10921937&dopt=Abstract">Blackwood
    et al 2000 </A>used analysis of RecA for the identification of <I>Mycobacterium</I>
    species
    <LI><A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=10758889&dopt=Abstract">Marechal
    et al 2000 </A>used phylogenetic analysis of RecA to study the relationships
    of <I>Frankia</I> and <I>Acidothermus</I>
  </UL>
</UL>

<H3>RecA-like genes in eukaryotes and Archaea</H3>

<UL>
  <LI>Archaea: <A HREF="http://www4.ncbi.nlm.nih.gov/htbin-post/Entrez/query?uid=8668545&form=6&db=m&Dopt=r">Sandler
  et al. (1996)</A> reported the cloning and characteriztion of RecA-like
  genes from a variety of Archaea.<BR>
  
  <LI>Eukaryotes: <A HREF="http://www.ncbi.nlm.nih.gov/htbin-post/Entrez/query?uid=9021132&form=6&db=m&Dopt=r">Stassen
  et al. (1997)</A> studied the evolution of RecA-like proteins (Rad51, DMC1,
  etc) in eukaryotes.
</UL>

<P><HR ALIGN=LEFT></P>

<ADDRESS>Email comments or suggestions to <A HREF="mailto:jeisen@leland.stanford.edu">jeisen@tigr.org</A></ADDRESS>

<P><HR ALIGN=LEFT NOSHADE><A HREF="http://www.tigr.org/~jeisen/">Return
to Jonathan A. Eisen's Home Page</A>
</BODY>
</HTML>
