<HTML>
<HEAD>
  <META NAME="GENERATOR" CONTENT="Adobe PageMill 2.0 Mac">
  <TITLE>Genetic Data Environment version 2.2 manual</TITLE>
</HEAD>
<BODY>

<H2>Genetic Data Environment version 2.2</H2>

<H3>Table of Contents</H3>

<UL>
  <LI><A HREF="#Introduction">Introduction</A>
  <LI><A HREF="#What's New">What's New for this Release</A>
  <LI><A HREF="#System Requirements">System Requirements</A>
  <LI><A HREF="#Note to Motif users">Note to Motif users</A>
  <LI><A HREF="#Installing the GDE">Installing the GDE</A>
  <LI><A HREF="#Using the GDE">Using the GDE</A>
  <LI><A HREF="#Data Types">Data Types</A>
  <LI><A HREF="#Menu Functions">Menu Functions</A>
  <LI><A HREF="#File menu">File menu</A>
  <LI><A HREF="#Edit menu">Edit menu</A>
  <LI><A HREF="#DNA/RNA menu">DNA/RNA menu</A>
  <LI><A HREF="#External Functions">External Functions</A>
  <LI>Bug reports/extensions
  <LI><A HREF="#Acknowledgments">Acknowledgments</A>
  <LI><A HREF="AppendixA.html">Appendix A</A>, File Formats
  <LI><A HREF="AppendixB.html">Appendix B</A>, Adding Functions
  <LI><A HREF="AppendixC.html">Appendix C</A>, External functions
  <LI><A HREF="#Copyright Notice">Copyright Notice</A>
</UL>

<H3><A NAME="Introduction"></A>Introduction</H3>

<P>The Genetic Data Environment is part of a growing set of programs for
manipulating and analyzing &quot;genetic&quot; data. It differs in design
from other analysis programs in that it is intended to be an expandable
and customizable system, while still being easy to use.</P>

<P>There are a tremendous number of publicly available programs for sequence
analysis. Many of these programs have found their way into commercial packages
which incorporate them into integrated, easy to use systems. The goal of
the GDE is to minimize the amount of effort required to integrate sequence
analysis functions into a common environment. The GDE takes care of the
user interface issues, and allows the programmer to concentrate on the analysis
itself. Existing programs can be tied into the GDE in a matter of hours
(or minutes) as apposed to days or weeks. Programs may be written in any
language, and still seamlessly be incorporated into the GDE.</P>

<P>These programs are, and will continue to be, available at no charge.
It is the hope that this system will grow in functionality as more and more
people see the benefits of a modular analysis environment. Users are encouraged
to make modifications to the system, and forward all changes and additions
to Steven Smith at smith@bioimage.millipore.com.</P>

<H3><A NAME="What's New"></A>What's New for this Release</H3>

<P>GDE 2.2 represents a maintainence release. Several small bugs have been
fixed, as well as new editing features and user interface elements. Also,
I have tried to update all of the contributed external programs to their
latest release. Updated programs include: <B>Phylip</B>, <B>Treetool</B>,
<B>LoopTool</B>, <B>Readseq</B>, <B>Blast</B>, and <B>Fasta</B>. Improved
versions of printing, and translate are included as well. As for new editing
features, a useful &quot;yanking&quot; feature has been added by Scott Ferguson
from Exxon Research, and the capability to export the colormap for a seqeunce
(see <A HREF="AppendixA.html">appendix A</A> and <A HREF="AppendixC.html">appendix
C</A>). Among the bugs fixed in this release are: <BR>
<BR>
</P>

<UL>
  <LI>Selection mask problems when exporting to Genbank (fixed in 2.1)
  <LI>Memory leaks (fixed in 2.1)
  <LI>Correct handling of circular sequences
  <LI>More liberal interpretation of Genbank formatted files. (not column
  dependent)
</UL>

<H3><A NAME="System Requirements"></A>System Requirements</H3>

<P>GDE 2.2 currently runs on the Sun family of workstations. This includes
the Sun3 and Sun4 (Sparcstation) systems. It was written in XView, and runs
on Suns using OpenWindows 3.0 or MIT's X Windows. It runs in both monochrome,
and color, and can be run remotely on any system capable of running X Windows
Release 4. You should have at least 15 meg of free disk space available.
The binary release for SparcStations was compiled under SunOS 4.1.2 and
Openwindows 3.0. <BR>
<BR>
We are also supporting a DECStation version of GDE. This is running under
XView 3.0/X11R5. We encourage interested people to port the programs to
their favorite Unix platform. There are informal ports to the SGI line of
unix machines.</P>

<H3><A NAME="Note to Motif users"></A>Note to Motif users</H3>

<P>GDE2.2 can be run using different window managers. The most common alternative
to olwm is the Motif window manager (mwm). The only problem in using another
window manager is that the status line is not displayed. We have added a
&quot;Message panel&quot; as an option under &quot;File-&gt;Properties&quot;
which displays all of the information contained on the status line. <BR>
<BR>
People using other window managers may also prefer using xterm, and xedit
as default terminals and file editors. This can be accomplished by replacing
all occurrences of 'shelltool' and 'textedit' with 'xterm -e' and 'xedit'
in the $GDE_HELP_DIR/.GDEmenus file.</P>

<H3><A NAME="Installing the GDE"></A>Installing the GDE</H3>

<P>Instructions for the source code release are included in the README.install
file. <BR>
<BR>
The binary installations consist of creating a GDE directory, such as /usr/local/GDE,
and un-taring the installation tarfile into the directory. If you are installing
the GDE for your own use, then you can simply make a GDE subdirectory. There
is no need to be superuser (root) to do the installation in your own directory.
For example:</P>

<UL>
  <LI>demo% mkdir /usr/local/GDE
  <LI>demo% cp GDE2.2.tar /usr/local/GDE
  <LI>demo% cd /usr/local/GDE
  <LI>demo% tar -xf GDE2.2.tar
</UL>

<P>After this, each new user will need to add two lines to their .cshrc
file so that they can find the gde programs and files.</P>

<UL>
  <LI>demo% cat &gt;&gt; ~/.cshrc
  <LI>set path = ($path /usr/local/GDE/bin)
  <LI>setenv GDE_HELP_DIR /usr/local/GDE/help/
  <LI>^D
</UL>

<P>You may wish to make a copy of the .GDEmenus file from the help directory
into your home directory. This is only necessary if you wish to modify your
menus. Copy the demo files from /usr/local/GDE/demo into your local directory,
and you are now ready to use the GDE. <BR>
<BR>
FastA and Blast need to have the properly formatted databases installed
in the $GDE_HELP_DIR under the directories FASTA/PIR, FASTA/GENBANK, BLAST/pir
BLAST/genbank. For FASTA, simply copy a version of PIR and Genbank into
the proper directory. Alternately, the PIR and GENBANK files can be symbolic
links to copies of Genbank held elsewhere on your system. You may need to
look at the .GDEmenus file in $GDE_HELP_DIR to verify that you are using
the same divisions for these databases. <BR>
<BR>
Blast installation involves converting PIR and GENBANK to a temporary FASTA
format (using pir2fasta and gb2fasta) and then using pressdb for nucleic
acid, and setdb for amino acid to reformat the databases again into blast
format. The .GDEmenus file is currently set up to search with blast using
the following databases: pir, genpept, genupdate, and genbank. If you wish
to divide these into subdivisions, then the .GDEmenus file will have to
be edited. <BR>
<BR>
The most up to date release of blast can be obtained via anonymous ftp to
<A HREF="ftp://ncbi.nlm.nih.gov/">ncbi.nlm.nih.gov</A>. The most recent
release of FASTA can be obtained via anonymous ftp to <A HREF="ftp://uvaarpa.virginia.edu/">uvaarpa.virginia.edu</A>.
It is strongly recommended that you retrieve these copies, and become familiar
with their setup. <HR ALIGN=LEFT NOSHADE></P>

<H3><A NAME="Using the GDE"></A>Using the GDE</H3>

<P>It is assumed that the user is familiar with the Unix, and OpenWindows/Xwindows
environments. It is also assumed that people running standard MIT X-Windows
will be using the OpenLook window manager (olwm). Other window managers
work with varied success. If you are not certain as to how your system is
set up, please contact your systems administrator.<BR>
<BR>
Once the window system has started, and a terminal window (xterm, shelltool
etc.) you can start up the GDE by typing:<B> gde tRNAs</B>. This should
load the sample data set tRNAs into GDE, and the following window should
appear: <BR>
<BR>
<IMG SRC="Fig1.gif" ALIGN="TOP" WIDTH="455" HEIGHT="198" NATURALSIZEFLAG=
"3"> <BR>
<BR>
This is the sequence alignment editor. It consists of a color alignment
display, a set of command menus, horizontal and vertical scroll bars to
navigate the alignment, a list of short sequence names (usually the LOCUS
of a Genbank entry), and a status line. The cursor is located in the upper
left corner. <BR>
</P>

<UL>
  <LI><B>Using the Mouse.</B> The mouse follow OpenLook standards for operation.
  The functions for each button are: <BR>
  <BR>
  <IMG SRC="Fig2.gif" ALIGN="TOP" WIDTH="257" HEIGHT="109" NATURALSIZEFLAG=
  "3"> <BR>
  <BR>
  The left mouse button is used for placing the cursor, selecting sequences
  by their short names, scrolling/paging, performing split screens, and resizing.
  The right button is used for pop up menus, and scrollbar menus. The middle
  button is used for extending a text selection. <BR>
  <BR>
  
  <LI><B>Cursor Movement.</B> The cursor can be moved using the arrow keys,
  or by clicking the mouse within a sequence. The cursors position is displayed
  on the status line in both sequence position and alignment column number.
  The right hand side of the status line shows the left and right column
  positions of the currently active display.<BR>
  <BR>
  Scrolling is controlled by the scrollbar elevator. By clicking (left mouse
  button) on one of the elevator arrows, the screen will scroll one character
  in that direction. By dragging the elevator center, the screen can be moved
  directly to any location. By clicking directly to one side of the elevator,
  the screen will scroll one full screen in that direction. And by clicking
  on the scrollbar anchor, the elevator will move to that anchor. Scrollbars
  also have menus associated with them giving other scroll options. Use the
  right mouse button to activate the menu. <BR>
  <BR>
  
  <LI><B>Selecting Sequences</B>. Sequence selection is necessary before
  most functions can be performed. Selecting sequences is accomplished by
  clicking or dragging (left button) over the short name associated with
  the sequence(s). The name of the sequence should become highlighted on
  the release of the mouse button. By holding down the shift key, you can
  toggle the selection on or off for any set of sequences. By clicking just
  to the right of any sequence short name, you will deselect all of them.
  <BR>
  <BR>
  
  <LI><B>Selecting Text.</B> Selecting text is accomplished in much the same
  way as selecting entire sequences. In the editing window, you can drag
  the mouse pointer over a rectangular region the select a block of text.
  By using the shift key (or the middle mouse button) you can adjust the
  selection to include other sequences, or other columns of text. If groups
  are enabled, GDE will automatically select all sequences in a group if
  any one sequence in a group is selected (See Sequence Editing). <BR>
  <BR>
  
  <LI><B>Sequence Protection.</B> All sequences can be individually protected
  against accidental modification. This is accomplished by selecting the
  set of sequences that you are interested in editing, and choosing the &quot;Set
  protections&quot; menu item under the File menu. Your choices are:<BR>
  <BR>
  
  <UL>
    <LI>Unambiguous modification ..... Changing/adding/deleting regular characters
    <LI>Ambiguous changes ..... Changing ambiguous codes ('N', 'X'...)
    <LI>Alignment modifications ..... Changing alignment gaps ('-', '~')
  </UL>
  <P><BR>
  
  <LI><B>Sequence Editing.</B> Sequences can be edited by simply typing to
  insert, and using the delete or backspace key to delete characters. Sequences
  must have the proper protections set to allow the type of modifications
  that you are attempting. The default protection level only allows modification
  to the alignment, but not to the sequences themselves. The Sun function
  keys, cut, copy and paste are used to edit selected text. Text selections
  work in rectangular (possibly disjointed) regions. You can cut or copy
  a block of sequence text, and paste it to a new cursor location using these
  three keys. <BR>
  <BR>
  
  <LI><B>Sequence Yanking.</B> Yanking referes to the &quot;pulling&quot;
  of a base to fill a gapped position like beads on an abacus. Place the
  cursor over a gap character, and type control-k to yank the character from
  the left into the current position. Type control-l to pull the character
  from the right. Repeat counts are honored (&quot;20 control-l&quot; will
  yank 20 characters from the right). <BR>
  <BR>
  
  <LI><B>Repeat Counts.</B> By typing a numeric value before an editing function
  you can insert, delete or move a number of characters at a time. The current
  repeat count is displayed on the status line, and can be cleared by clicking
  the left mouse button in the alignment window. In order to insert twenty
  gaps into a sequence, one would type &quot;20-&quot;. In order to move
  down five sequences, one would type &quot;5&Oslash;&quot;. This works with
  all sequence types, however the meta (diamond) key must be held down when
  the cursor is in a text or mask sequence. This is because numbers are valid
  characters in these sequences, and would otherwise be confused with repeat
  counts. <BR>
  <BR>
  
  <LI><B>Split Screen.</B> Split screen editing allows the viewing one region
  while editing another. This is very useful for aligning &quot;downstream&quot;
  regions by editing &quot;upstream&quot;. The alignment window can be split
  horizontally into two or more windows into the alignment. These windows
  scroll independently of each other both horizontally and vertically. The
  short names displayed to the left of the alignment correspond to the window
  that was last scrolled or edited. Care should be taken in any modifications
  done in this mode so that edits are performed on the correct sequence.
  To avoid confusion during split screen operations, the vertical scroll
  bars may be locked so that all windows scroll together. <BR>
  <BR>
  <IMG SRC="Fig4.gif" ALIGN="TOP" WIDTH="250" HEIGHT="153" NATURALSIZEFLAG=
  "3"> <BR>
  <BR>
  In order to split a window into two views, grab (left button) the left
  or right anchor (small rectangle) at either end of the horizontal scrollbar
  and drag to the middle of the window. This should split the window into
  two views. To join two views, place the mouse pointer on the horizontal
  scroll bar use the menu (right button). The views are NOT two copies of
  the alignment. Changes in one window are reflected in the other. Users
  should not be confused by this fact. <BR>
  <BR>
  
  <LI><B>Sequence Grouping.</B> Sequences can be grouped for editing functions.
  This is very helpful when trying to adjust several sub alignments. When
  grouped, all sequences within a group will be affected by editing in any
  member of the group. All sequences within a group must have protections
  set to allow modification before any one will be modified. In order to
  group sequences, select the names of the sequences that should fall within
  a group, and select Group under the Edit menu. A number will be placed
  at the left of the sequence representing its assigned group number. To
  any sequence or sequences, the user selects those sequences and uses the
  Ungroup command under the Edit menu. <BR>
  <BR>
  
  <LI><B>Special keys.</B> There are also a few special function keys used
  in the GDE. Some functions have meta key equivalences so that they can
  be called from the keyboard, instead of by the menu system. The &quot;meta&quot;
  key is a standard property of X windows, and may be remapped to a different
  key symbol for different keyboards. For example, meta on Sun workstations
  is represented with a &brvbar;, where on a Macintosh running MacX it might
  be the &quot;apple&quot; key. The operation of the key is the same as the
  control or shift key, it is held down while pressing the second key in
  the sequence. <BR>
  <BR>
  Cut text, copy text and paste text are mapped to the Openlook equivalent
  keys (L10, L6, and L8 on Sun keyboards). Other meta keys are defined in
  the .GDEmenus file, and may be changed to suit your preferences.
</UL>

<P><HR ALIGN=LEFT NOSHADE></P>

<H2><A NAME="Data Types"></A>Data Types</H2>

<P>The GDE supports several data types. The data types supported in 2.2
are DNA, RNA, protein (single letter codes), mask sequence, and text.</P>

<UL>
  <LI><B>DNA and RNA.</B> Nucleic acid sequences are tightly type cast, and
  can contain any IUPAC code (ACGTUM RSVWYHKDBN) as well as two alignment
  gap characters ('~' and '-'). Some keys are remapped to fit IUPAC codes.
  For example, 'X' is mapped to 'N'. All nonstandard characters get mapped
  to the alignment gap '-'. Upper and lower case are both supported, and
  the T/U characters are mapped based on whether you are working with DNA
  or RNA. The color coding for DNA and RNA is identical. The color for ambiguous
  characters, and for alignment gaps is grey. <BR>
  <BR>
  
  <LI><B>Amino Acid Sequence.</B> Amino acid sequences are loosely type cast,
  and can contain any valid ASCII character. The results of analysis on nonstandard
  characters is not guaranteed. The color for nonstandard amino acid characters,
  and for alignment gaps is grey. <BR>
  <BR>
  
  <LI><B>Text Sequence.</B> Any valid ASCII printable character can be entered
  into a text sequence. Care should be taken with using space characters,
  as these will only be saved properly in Genbank format, and not in flat
  file format. The characters @#% and &quot; should be avoided as well, as
  these can confuse the reading of flat files if saved in that format. <BR>
  <BR>
  
  <LI><B>Mask Sequence.</B> Mask sequence is identical to text sequence with
  the following exceptions. Mask sequence can have the ability (function
  dependent) of masking out positions in an alignment for analysis. If a
  mask sequence is selected along with some other sequence(s) for an analysis
  function that permits masking, then all columns that contain a '0' in the
  mask sequence will be ignored by the function. The mask itself would not
  be passed to the analysis function either. Some functions allow masking,
  some do not. Refer to the instruction page for each function to see whether
  or not it supports sequence masking. <BR>
  <BR>
  
  <LI><B>Color Masks.</B> Color masks give color to a sequence on a position
  by position basis. Individual sequences can have color masks attached to
  them, or one color mask can be used for an entire alignment. Color masks
  are generated externally by some analysis functions, and are then passed
  back to the GDE. The file format for a colormask is described in <A HREF=
  "AppendixA.html">Appendix A</A>.
</UL>

<P><HR ALIGN=LEFT NOSHADE></P>

<H2><A NAME="Menu Functions:"></A>Menu Functions</H2>

<UL>
  <LI><A NAME="File menu"></A><B>File menu</B>. The GDE has several built-in
  menu functions under the File and Edit menus. These functions are unique
  in that they are part of the primary display editor, and are not described
  in the .GDEmenus file. <BR>
  <BR>
  
  <UL>
    <LI><B>Open...</B> Selecting this will bring up the open file dialog box.
    Users can scroll through a list of files in the current directory, move
    up and down the directory tree, and open any individual data file. The
    sequence data in that file is loaded into the current editing window below
    any existing sequences. The open command will open any Genbank formatted
    file, or a GDE flat file. <BR>
    <BR>
    
    <LI><B>Save as...</B> This function will save the entire alignment to a
    specified file in either Genbank or flat file format. The file will be
    saved in the local directory unless a relative or absolute path is specified.
    <BR>
    <BR>
    
    <LI><B>Properties...</B> Properties controls the display settings. Those
    settings include character size, color type, and insert direction. The
    screen can also be inverted, vertical scroll lock and keyboard clicks (tactile
    feedback) can be turned on or off. Vertical scrollbar lock will cause all
    split views to scroll together in the vertical direction. <BR>
    <BR>
    <IMG SRC="Fig5.gif" ALIGN="TOP" WIDTH="136" HEIGHT="135" NATURALSIZEFLAG=
    "3"> <BR>
    <BR>
    
    <LI><B>Protections...</B> This will display, and then set the default protections
    for all selected sequences. If two or more of the sequences differ in their
    current protection settings, a warning message will appear in the protection
    dialog box. The protections currently available are alignment gap protection,
    ambiguous character protection, unambiguous character protection, and translation
    protection. <BR>
    <BR>
    <IMG SRC="Fig6.gif" ALIGN="TOP" WIDTH="82" HEIGHT="80" NATURALSIZEFLAG=
    "3"> <BR>
    <BR>
    
    <LI><B>Get info...</B> This option allows the viewing and setting of attributes
    associated with each individual sequence. These attributes include short
    name, full name, description, author, comments, and the sequence type.
    The attributes loosely correspond to fields in a Genbank entry. Comments
    can be included for each sequence in the comments field. <BR>
    <BR>
    <IMG SRC="Fig7.gif" ALIGN="TOP" WIDTH="257" HEIGHT="179" NATURALSIZEFLAG=
    "3"> <BR>
    <BR>
    
  </UL>
  <LI><A NAME="Edit menu"></A><B>Edit menu</B> <BR>
  
  <UL>
    <LI><B>Select All.</B> Selects all sequences. This is helpful when you
    have several dozen sequences. <BR>
    <BR>
    
    <LI><B>Select by name ...</B> Select all sequences containing a given string
    in their short names field. No wild cards are allowed, and only selecting
    is allowed, not de-selecting. The search is started when the Return key
    is pressed, and multiple searches can be accumulated. Press Done when finished.
    <BR>
    <BR>
    
    <LI><B>Cut/Copy/Paste sequences.</B> Cut copy and paste are primarily useful
    for reordering sequences, and for making duplicate copies of a given sequence.
    They do not pass information to other programs. This capability will be
    implemented in a later release. Cut and copy will place the selected sequences
    on an internal clipboard. They can then be pasted back into the top of
    editing window (default) or under the last selected sequence. <BR>
    <BR>
    
    <LI><B>Group/Ungroup. </B>Assign a group number to the selected sequences.
    Edit operations in any one sequence within the group will be propagated
    to all within the group. Sequence protections from one group are also imposed
    upon all other sequence in the given group. If a given operation is illegal
    in one sequence in a group (i.e. alignment modification) then it will not
    work in any of the sequences in that group. Ungroup will remove the selected
    sequences from a given group. <BR>
    <BR>
    
    <LI><B>Compress.</B> Compress will remove gap characters from the selected
    sequences. The user has the option of removing all gaps, or simply all
    columns containing nothing but gaps. This is useful for minimizing the
    length of a subalignment. <BR>
    <BR>
    
    <LI><B>Reverse Sequence.</B> Reverses the selected sequences. Alignment
    gaps are reversed as well. The selected sequences will remain aligned after
    reversal.
  </UL>
  <P><BR>
  <BR>
  
  <LI><A NAME="DNA/RNA menu"></A><B>DNA/RNA menu</B> <BR>
  <BR>
  
  <UL>
    <LI><B>Complement Sequence.</B> Converts DNA/RNA into its complement strand
    (keeping full IUPAC ambiguity). This function has no effect on text, protein,
    or mask sequence. Note that this function does not produce the reverse
    strand of DNA but merely converts A&lt;-&gt;T and G&lt;-&gt;C. If the reverse
    strand is needed, remember to Complement and Reverse the sequence (Edit
    menu).
  </UL>
</UL>

<P><HR ALIGN=LEFT NOSHADE></P>

<H2><A NAME="External Functions"></A>External Functions</H2>

<P>See appendix C for a full description of functions supported in GDE 2.2
All external functions are described in the configuration file .GDEmenus.
Here is a brief description of some of the basic functions included.</P>

<UL>
  <LI><B>File menu</B> <BR>
  <BR>
  
  <UL>
    <LI><B>New Sequence</B> (meta n). Create a new sequence. Prompts for sequence
    type, and short name. <BR>
    <BR>
    
    <LI><B>Import foreign format/Export foreign format.</B> Load and save sequences
    using Readseq by Don Gilbert <A HREF="AppendixC.html">(see Appendix C)</A>.
    <BR>
    <BR>
    
    <LI><B>Save Selection.</B> Save the currently selected sequences in a specified
    file. <BR>
    <BR>
    
    <LI><B>Pretty print.</B> Print using the sequence formatter supplied by
    Readseq. <BR>
    <BR>
    
    <LI><B>Print Selection.</B> Print the selected sequences to the chosen
    printer. This function supports the Unix command enscript as well as lpr.
    The .GDEmenus file may need to be modified to add the names of local printers
    to the printer list.
  </UL>
  <P><BR>
  <BR>
  
  <LI><B>Edit menu. <BR>
  <BR>
  </B>
  <UL>
    <LI><B>Sort...</B> Sort the selected sequences by a primary and secondary
    key. Pass the new order to a new GDE window. <BR>
    <BR>
    
    <LI><B>Extract.</B> Extract the selected sequences into a new window.
  </UL>
  <P><BR>
  <BR>
  
  <LI><B>DNA/RNA Menu.</B> <BR>
  <BR>
  
  <UL>
    <LI><B>Translate.</B> Translate the selected sequences from DNA/RNA to
    Amino acid. The user can specify the desired reading frame, and the minimum
    open reading frame (stop codon to stop codon) to translate. The user can
    also choose between single letter code and triple letter codes. There is
    also an option to allow each ORF to be entered as a seperate sequence.
    <BR>
    <BR>
    
    <LI><B>Dot plot.</B> Display a dotplot identity matrix for the selected
    sequence(s). If only one sequence is selected, then the dotplot is a self
    comparison. If two or more sequences are selected, then the first two sequences
    are compared. <BR>
    <BR>
    
    <LI><B>Clustal Align.</B> Align the selected sequences using the clustalv
    algorithm by Des Higgins.<A HREF="AppendixC.html">(See Appendix C)</A>
    <BR>
    <BR>
    
    <LI><B>Find All</B> (meta f). Search and highlight the selected sequences
    for a given substring. A specified percent of mismatching can also be allowed.
    <BR>
    <BR>
    
    <LI><B>Variable Positions.</B> The selected sequences are scored column
    by column for conservation. The result is returned as a grey scale alignment
    color mask. This can be useful in selecting PCR primers. <BR>
    <BR>
    
    <LI><B>Sequence Consensus.</B> Return the consensus for the selected sequences.
    This can either be a majority consensus, or an ambiguity consensus using
    IUPAC coding. <BR>
    <BR>
    
    <LI><B>Distance Matrix.</B> Calculate a distance matrix for the selected
    sequences.<A HREF="AppendixC.html">(See Appendix C)</A> <BR>
    <BR>
    
    <LI><B>MFOLD.</B> Fold the selected sequences using MFOLD by Michael Zuker.
    The resulting structure is returned as a nested bracket ('[]') representation
    of the secondary structure.<A HREF="AppendixC.html">(See Appendix C)</A>
    <BR>
    <BR>
    
    <LI><B>Draw Secondary Structure.</B> Draw the selected sequence using the
    proposed secondary structure. Both the secondary structure prediction,
    and the RNA sequence should be selected before calling this function. The
    drawing program is LoopTool.<A HREF="AppendixC.html">(See Appendix C)</A>
    <BR>
    <BR>
    
    <LI><B>Highlight Helix.</B> Show all violations to a proposed RNA secondary
    structure. The secondary structure represented must be selected, as well
    as the aligned sequences to be tested. The selected sequences will then
    be colored according to whether or not they support the proposed 2&deg;
    structure. Standard Watson/Crick paring will be colored dark blue, G-U
    paring will be colored light blue, mismatches will be colored gold, and
    pairng to gaps will be red. <BR>
    <BR>
    
    <LI><B>Blastn/BlastX.</B> Search the selected sequence (select only one)
    against a given database with the BLAST searching tool written by Altschul,
    Gish, Miller, Myers, and Lipman. Blastn searches DNA against DNA databases,
    blastx searches DNA against AA databases by translating the sequence in
    all six reading frames.<A HREF="AppendixC.html">(See Appendix C)</A> <BR>
    <BR>
    
    <LI><B>FastA.</B> Search the selected sequence (select only one) against
    a given database using the FASTA similarity search program written by Pearson
    and Lipman.<A HREF="AppendixC.html">(See Appendix C)</A>
  </UL>
  <P><BR>
  <BR>
  
  <LI><B>Protein Menu.</B> <BR>
  <BR>
  
  <UL>
    <LI><B>Clustal Align.</B> Align the selected amino acid sequences using
    the clustal algorithm. <A HREF="AppendixC.html">(See Appendix C)</A> <BR>
    <BR>
    
    <LI><B>Blastp, Tblastn, Blast3.</B> Search the selected sequence (select
    only one) against a given database with the BLAST searching tool written
    by Altschul, Gish, Miller, Myers, and Lipman. Blastp searches AA against
    AA databases, tblastn searches AA against DNA databases by translating
    the database in all six reading frames. Blast3 finds three way alignments
    that are could not be found with only pairwise comparisons.<A HREF="AppendixC.html">(See
    Appendix C)</A>
  </UL>
  <P><BR>
  <BR>
  
  <LI><B>Sequence Management Menu.</B> <BR>
  <BR>
  
  <UL>
    <LI><B>Assemble contigs.</B> Assemble the selected sequences into contigs
    using the program CAP (Contig Assemble Program) written by Xiaoqiu Huang.
    The resulting sequences are returned to the current GDE window, and they
    are grouped into contigs. The user can then sort the sequences by group,
    and offset to produce an ordered list of the contigs.<A HREF="AppendixC.html">(See
    Appendix C)</A> <BR>
    <BR>
    
    <LI><B>Strategy view.</B> Pass out the selected sequences to StratView.
    This program will display contigs in a greatly reduced line drawing. This
    is very useful for large contigs. <BR>
    <BR>
    
    <LI><B>Restriction sites.</B> Search the selected sequences for the restriction
    enzymes specified in the given enzyme file. The restriction sites are then
    colored by enzyme.
  </UL>
  <P><BR>
  <BR>
  
  <LI><B>Phylogeny menu.</B> <BR>
  <BR>
  
  <UL>
    <LI><B>DeSoete Tree fit.</B> Calculate a phylogenetic tree using a least
    squares fitting algorithm on a distance matrix calculated from the selected
    sequences. The results can then be passed on to treetool for display and
    manipulation.<A HREF="AppendixC.html">(See Appendix C)</A> <BR>
    <BR>
    
    <LI><B>Phylip 3.5.</B>Pass the selected data to on of the treeing programs
    in Phylip, written by Joe Felsenstein. The chosen phylip program is started
    in it's own window, with the selected sequences already loaded.<A HREF=
    "AppendixC.html">(See Appendix C)</A>
  </UL>
</UL>

<H2>Citation of work</H2>

<P>We ask that any published work using any of the external functions in
GDE cite the appropriate authors. Please <A HREF="AppendixC.html">see Appendix
C</A> for references.</P>

<H2><A NAME="Acknowledgments"></A>Acknowledgments</H2>

<P>I would like to thank the following people for their input and assistance
and code used in the development of the GDE: Carl Woese, Gary Olsen and
Mike Maciukenas at University of Illinois Dept of Microbiology, Ross Overbeek
at Argonne National Laboratories,Walter Gilbert, Patrick Gillevet, Chunwei
Wang, Susan Russo and Erik Bunce at the Harvard Genome Laboratory. I would
also like to personally thank the following people for their permission
to include their software with this release of GDE.</P>

<UL>
  <LI>Tim Littlejohn
  <LI>Scott Ferguson
  <LI>Brian Fristensky
  <LI>Des Higgins
  <LI>David Lipman and the group at NCBI
  <LI>William Pearson
  <LI>Don Gilbert
  <LI>Xiaoqui Huang
  <LI>Joe Felsenstein
  <LI>Michael Zuker
  <LI>Geert DeSoete
</UL>

<P>Many thanks to all the people who have directly and indirectly helped
with the ongoing support of GDE. It is only by the generosity of these people
that GDE has been successful.</P>

<H2><A NAME="Copyright Notice"></A>Copyright Notice</H2>

<P>The Genetic Data Environment (GDE) software and documentation are not
in the public domain. Portions of this code are owned and copyrighted by
the The Board of Trustees of the University of Illinois and by Steven Smith.
External functions used by GDE are the proporty of, their respective authors.
This release of the GDE program and documentation may not be sold, or incorporated
into a commercial product, in whole or in part without the expressed written
consent of the University of Illinois and of its author, Steven Smith. <BR>
<BR>
All interested parties may redistribute the GDE as long as all copies are
accompanied by this documentation, and all copyright notices remain intact.
Parties interested in redistribution must do so on a non-profit basis, charging
only for cost of media. Modifications to the GDE core editor should be forwarded
to the author Steven Smith. External programs used by the GDE are copyright
by, and are the property of their respective authors unless otherwise stated.
<BR>
<BR>
While all attempts have been made to insure the integrity of these programs:</P>

<H3>Disclaimer</H3>

<P>THE UNIVERSITY OF ILLINOIS, HARVARD UNIVERSITY AND THE AUTHOR, STEVEN
SMITH GIVE NO WARRANTIES, EXPRESSED OR IMPLIED FOR THE SOFTWARE AND DOCUMENTATION
PROVIDED, INCLUDING, BUT NOT LIMITED TO WARRANTY OF MERCHANTABILITY AND
WARRANTY OF FITNESS FOR A PARTICULAR PURPOSE. User understands the software
is a research tool for which no warranties as to capabilities or accuracy
are made, and user accepts the software &quot;as is.&quot; User assumes
the entire risk as to the results and performance of the software and documentation.
The above parties cannot be held liable for any direct, indirect, consequential
or incidental damages with respect to any claim by user or any third party
on account of, or arising from the use of software and associated materials.
This disclaimer covers both the GDE core editor and all external programs
used by the GDE.
</BODY>
</HTML>
